packages/trpc/server/routers/viewer/teams/delete.handler.ts

```diff
@@ -23,25 +23,27 @@ export const deleteHandler = async ({ ctx, input }: DeleteOptions) => {
 
   if (IS_TEAM_BILLING_ENABLED) await cancelTeamSubscriptionFromStripe(input.teamId);
 
-  // delete all memberships
-  await prisma.membership.deleteMany({
-    where: {
-      teamId: input.teamId,
-    },
+  await prisma.$transaction(async (tx) => {
+    // delete all memberships
+    await tx.membership.deleteMany({
+      where: {
+        teamId: input.teamId,
+      },
+    });
+
+    const deletedTeam = await tx.team.delete({
+      where: {
+        id: input.teamId,
+      },
+    });
+
+    const deletedTeamMetadata = teamMetadataSchema.parse(deletedTeam.metadata);
+
+    if (deletedTeamMetadata?.isOrganization && deletedTeam.slug) deleteDomain(deletedTeam.slug);
+
+    // Sync Services: Close.cm
+    closeComDeleteTeam(deletedTeam);
```

Comment:
Third party services typically shouldn’t be wrapped in transactions like this. Was this intentional?